agent/csi/libstorage/csi_service_libstorage_idemp.go

```diff
@@ -118,26 +119,35 @@ func (d *driver) IsControllerPublished(
 		return nil, nil
 	}
 
-	d.pubInfoRWL.RLock()
-	defer d.pubInfoRWL.RUnlock()
-
 	pvi := &csi.PublishVolumeInfo{
 		Values: map[string]string{
 			"encrypted": fmt.Sprintf("%v", vol.Encrypted),
 		},
 	}
 
-	if d.pubInfo[idVal] == nil {
-		// We don't have cached PublishVolumeInfo details, so we don't
-		// know what the state is, and cannot return an idempotent response
-		// This is where "Andrew was right" and it would be useful to know
-		// the full CSI method name. If we are doing a publish, we would
-		// return false, if it we are doing an unpublish, we would return
-		// we would return true to defer to the bridge
-		pvi.Values["token"] = ""
-	} else {
-		pvi.Values["token"] = d.pubInfo[idVal].attToken
+	// We know that the volume is attached, we need to go get it's token
+	// via local devices
+	ldOpts := &apitypes.LocalDevicesOpts{
+		Opts:     apiutils.NewStore(),
+		ScanType: apitypes.DeviceScanQuick,
+	}
+	devs, err := d.client.Executor().LocalDevices(d.ctx, ldOpts)
+	if err != nil {
+		return nil, errUnableToGetLocDevs
 	}
+
+	for k, v := range devs.DeviceMap {
+		if v == vol.Attachments[0].DeviceName {
```

Comment:
Should we range the attachments here too? ie do we care that the bridge may not do multi-attach?

Reply:
Hi @clintkitson,

I see your point, except we've already declared that a COs cannot access a single Node as both CSI *and* Docker Volume API. It's one or the other since Docker's model is sufficiently different.

Reply:
Thanks

Reply:
FWIW, I've no problem adding a `// TODO` here and attempting to come back later to make the Docker bridge able to tolerate multiple mount points, but at the moment this is just one of a few locations where single-attachment is explicitly the case.

Reply:
> a few locations

In the Docker bridge that is.

Reply:
Hi @clintkitson,

One last thing, this is data derived from libStorage (its in the idemp provider for the LS bridge). Thus the LS driver would also have to provide support for multiple attachments. Most of ours do not.